server/src/main/java/io/druid/metadata/SQLMetadataStorageActionHandler.java

```diff
@@ -206,306 +200,265 @@ public Boolean withHandle(Handle handle) throws Exception
+                        entryTable
+                    )
+                ).bind("start", timestamp.toString())
+                .map(
+                    new ResultSetMapper<StatusType>()
                     {
-                      try {
-                        return jsonMapper.readValue(
-                            r.getBytes("status_payload"),
-                            statusType
-                        );
-                      }
-                      catch (IOException e) {
-                        log.makeAlert(e, "Failed to parse status payload")
-                           .addData("entry", r.getString("id"))
-                           .emit();
-                        throw new SQLException(e);
+                      @Override
+                      public StatusType map(int index, ResultSet r, StatementContext ctx) throws SQLException
+                      {
+                        try {
+                          return jsonMapper.readValue(
+                              r.getBytes("status_payload"),
+                              statusType
+                          );
+                        }
+                        catch (IOException e) {
+                          log.makeAlert(e, "Failed to parse status payload")
+                             .addData("entry", r.getString("id"))
+                             .emit();
+                          throw new SQLException(e);
+                        }
                       }
                     }
-                  }
-              ).list();
+                ).list();
+          }
         }
-      }
     );
   }
 
   public boolean addLock(final String entryId, final LockType lock)
   {
-    return retryingHandle(
-      new HandleCallback<Boolean>()
-      {
-        @Override
-        public Boolean withHandle(Handle handle) throws Exception
+    return connector.retryWithHandle(
+        new HandleCallback<Boolean>()
         {
-          return handle.createStatement(
-                      String.format(
-                          "INSERT INTO %1$s (%2$s_id, lock_payload) VALUES (:entryId, :payload)",
-                          lockTable, entryTypeName
-                      )
-                  )
-                       .bind("entryId", entryId)
-                       .bind("payload", jsonMapper.writeValueAsBytes(lock))
-                       .execute() == 1;
+          @Override
+          public Boolean withHandle(Handle handle) throws Exception
+          {
+            return handle.createStatement(
+                String.format(
+                    "INSERT INTO %1$s (%2$s_id, lock_payload) VALUES (:entryId, :payload)",
+                    lockTable, entryTypeName
+                )
+            )
+                         .bind("entryId", entryId)
+                         .bind("payload", jsonMapper.writeValueAsBytes(lock))
+                         .execute() == 1;
```

Comment:
I think this is okay the way it is.

The reason is that locks are similar to logs in that the primary key is an auto_increment, so the worst that will happen here is a double-insert. That should be fine, since the restore process (syncing storage -> taskLockbox) only adds each taskId once to each lock, and the release process (metadata task storage removeLock) will remove all locks with a particular task id, so nothing should get out of whack.